doc/source/security/encrypting-binlogs.rst

```diff
@@ -0,0 +1,77 @@
+
+After you have enabled the ``binlog_encryption`` option and the keyring is available, you can encrypt new binary logs and relay logs on disk. Only the data content is encrypted.
+
+In replication, the master sends the stream of decrypted binary log events to a slave, in transport the data is encrypted by SSL connections. Master and slaves use separate keyring storages and are able to use different keyring plugins.
+
+When an encrypted binary log is dumped, and the operation involves decryption, and done using ``mysqlbinlog`` with ``--read-from-remote-server`` option.
+
+.. note::
+
+    The ``--read-from-remote-server`` option only applies to the binary logs. Encrypted relay logs can not be dumped or decrypted with this option.
+
+Attempting binlog encryption without the keyring generates a MySQL error.
+
+The Binary log encryption uses two tiers:
+
+    * File password
+
+    * Binary log encryption key
+
+The file password encrypts the content of a single binary file or relay log file. The binary log encryption key is used to encrypt the file password and is stored in the keyring.
+
+Enabling Binary Log Encryption
+-------------------------------
+
+To enable the ``binlog_encryption`` option you must set the option in a startup configuration file, such as the my.cnf file.
+
+.. code-block:: MySQL
+
+    binlog_encryption=ON
+
+Verifying the Encryption Setting
+----------------------------------
+
+You can verify if the binary log encryption option is enabled, you can run the following statement:
+
+.. code-block:: MySQL
+
+  mysql>SHOW BINARY LOGS;
+
+  +-------------------+----------------+---------------+
+  | Log_name          | File_size      | Encrypted     |
+  +-------------------+----------------+---------------+
+  | binlog.00011      | 72367          | No            |
+  | binlog:00012      | 71503          | No            |
+  | binlog:00013      | 73762          | Yes           |
+  +-------------------+----------------+---------------+
+
+A file has a 512 bytes header.
+
+.. seealso::
+
+  |MySQL| Documentation:
+  - `Encrypting Binary Log Files and Relay Log Files <https://dev.mysql.com/doc/refman/8.0/en/replication-binlog-encryption.html>`__
+
+Upgrading from |Percona Server| 8.0.15-5 to any Higher Version
+----------------------------------------------------------------
+
+Starting from release :rn:`8.0.15-5`, |Percona Server| uses the upstream implementation of binary log encryption. The variable encrypt-binlog is removed and the related command line option opt.encrypt-binlog is not supported. It is important to remove the encrypt-binlog variable from your configuration file before you attempt to upgrade either from another release in the |Percona Server| |version| series or from |Percona Server| 5.7. Otherwise, a server boot error will be produced reporting an unknown variable. The implemented binary log encryption is compatible with the old
```

Comment:
This `opt.encrypt-binlog` will be a link or something? It looks strange as cleartext.

Reply:
Reviewed the text with the 8.0.16 text. Updated to match that text.